assessment/report_generator.go

```diff
@@ -15,17 +15,147 @@
+	codeChangeEffort  string
+	codeImpactedFiles string
+	codeSnippets      string
+}
+
+func GenerateReport(dbName string, assessmentOutput utils.AssessmentOutput) {
 	//pull data from assessment output
 	//Write to report in require format
 	//publish report locally/on GCS
 	logger.Log.Info(fmt.Sprintf("%+v", assessmentOutput))
+
+	f, err := os.Create(dbName + "_schema.txt")
+	if err != nil {
+		logger.Log.Error(fmt.Sprintf("Can't create schema file %s: %v", dbName, err))
+		return
+	}
+	defer f.Close()
+
+	w := csv.NewWriter(f)
+	w.Comma = '|'
+	w.UseCRLF = true
+
+	w.WriteAll(generateSchemaReport(assessmentOutput))
+
 	logger.Log.Info("completed publishing sample report")
-	return
+}
+
+func generateSchemaReport(assessmentOutput utils.AssessmentOutput) [][]string {
+	var records [][]string
+
+	headers := getHeaders()
+
+	records = append(records, headers)
+
+	schemaReportRows := convertToSchemaReportRows(assessmentOutput)
+	for _, schemaRow := range schemaReportRows {
+		var row []string
+		row = append(row, schemaRow.element)
+		row = append(row, schemaRow.elementType)
+		row = append(row, schemaRow.sourceDefinition)
+		row = append(row, schemaRow.targetName)
+		row = append(row, schemaRow.targetDefinition)
+		row = append(row, schemaRow.dbChangeEffort)
+		row = append(row, schemaRow.dbChangeType)
+		row = append(row, schemaRow.codeChangeEffort)
+		row = append(row, schemaRow.codeChangeType)
+		row = append(row, schemaRow.codeImpactedFiles)
+		row = append(row, schemaRow.codeSnippets)
+
+		records = append(records, row)
+	}
+
+	return records
+}
+
+func getHeaders() []string {
+	headers := []string{
+		"Element",
+		"Element Type",
+		"Source Definition",
+		"Target Name",
+		"Target Definition",
+		//DB
+		"DB Change Type",
+		"DB Change Effort",
+		//CODE
+		"Code Change Type",
+		"Code Change Effort",
+		"Impacted Files",
+		"Related Code Snippets",
+	}
+	return headers
+}
+
+func convertToSchemaReportRows(assessmentOutput utils.AssessmentOutput) []SchemaReportRow {
+	rows := []SchemaReportRow{}
+
+	//Populate table info
+	for _, tableName := range assessmentOutput.SchemaAssessment.TableNames {
+		row := SchemaReportRow{}
+		row.element = tableName
+		row.elementType = "Table"
+		row.sourceDefinition = "N/A" //Todo - get table definition
+
+		row.targetName = "N/A"       // Get from spanner table def
+		row.targetDefinition = "N/A" // Get from spanner table def
+
+		row.dbChangeEffort = "Automatic"
+		row.dbChangeType = "None"
+
+		//Populate code info
+		rows = append(rows, row)
+	}
+
+	//Populate column info
+	for tableName, columnNames := range assessmentOutput.SchemaAssessment.ColumnNames {
+		for _, columnName := range columnNames {
+			row := SchemaReportRow{}
+			columnDefinition := assessmentOutput.SchemaAssessment.ColumnAssessmentOutput[columnName]
+			row.element = tableName + "." + columnName
+			row.elementType = "Column"
+			row.targetDefinition = columnDefinitionToString(columnDefinition)
+
```

Comment:
we need to add source definition, dbchangeeffort and dbchangeType here as well

Reply:
added